### PR TITLE
feat(chart-builder): match PanelEditModal to Figma 12:73

### LINKED ISSUE #438: Refresh Chart builder to Figma 12:73
What to build: Chart builder matches Refresh Figma Chart builder 12:73. Dark+gold. Potato: the builder surface for adding/editing a panel chart — not the agent dock, not MCP/Plugins.

Blocked by: None. Disjoint from Agent dock files. Parallel OK.

Figma: https://www.figma.com/design/39K1t4QzABolqRVKk7OUxg?node-id=12-73 page Refresh. Parent #432.

Own roughly: chart-builder / panel-editor UI under frontend (new or existing panel-create/edit path). Do not touch agent-dock layout, Explore query-editor chrome already shipped in #434, or Dashboard chrome already shipped in #433 unless a tiny shared token import is required.

Login + empty chrome stay on Proposals. Agent is a dock (separate ticket). Do not invent tokens.

Prove: chart builder at ace.janhoon.app (or PR preview) matches frame 12:73 for the default loaded/create state. Specs cover the new surface. Potato.

Poteto-mode. Own files only. Ready PR, no merge. Model grok-4.6, fast=false.

### PR DESCRIPTION
**Pull request by janhoon**
## Why

Chart builder create state did not match Refresh Figma 12:73. Add panel is one query, one chart, then pin. This PR restyles that editor only.

Closes #438. Parent #432 stays open.

## Scope

`PanelEditModal` is the Figma Main layout for **create** of chart-builder types (`line_chart` / `stat` / `table`). Agent dock, Explore, and dashboard chrome are untouched.

Designer notes for 12:73, already matching:

- Left config column is 360px. Preview pane fills the rest. Below ~425px the columns stack so the editor is not clipped.
- Type chips. Selected chip is a 1px primary accent stroke, not a gold fill.
- Pin CTA is ink-on-gold. Primary fill, `--primary-foreground` (`#0B0D0F` per DESIGN.md Selection Color). Never white on gold.
- Chart itself has no axes, legend, or grid. Caption is `No decoration. The chart is the panel.`
- Preview bars use the visualization palette default (`--color-viz-0` / Steel Blue), not brand primary.

Also in the builder: title `Add panel`, subtitle `One query. One chart. Then pin.`, default query `rate(http_requests_total[5m])`. Pin copy is `Pin to Victoria / prod` only when the selected datasource is actually Victoria/prod; otherwise it uses the datasource name or `Pin panel`.

## CI Jedi round 1 (behavior-preserving)

- Title fallback (`resolveChartBuilderTitle`) is **create-only**. Empty title on edit shows `Title is required` and does not call `updatePanel`.
- Four-chip chrome is gated to create / supported chart-builder types. Editing gauge, pie, logs, traces, text, etc. keeps a visible panel-type select and an honest preview label (never “Time series”).
- Heatmap chip stays in the Figma strip but is disabled until the registry marks it `ready`.
- Datasource picker is visible whenever datasources exist and the panel needs one.
- Close and Cancel are visible in the config column. Escape still dismisses.

## CI Jedi round 2 (behavior-preserving)

- Edit PromQL path restores `QueryBuilder`. Create keeps the Figma bare query input.
- Edit save CTA is `Save Changes`. Pin copy is create-only.
- Editing a chart-builder type keeps the chip strip **and** a visible full type select so authors can migrate to gauge/logs/etc. Create still hides the select behind chips.
- Preview bar strip fills remaining pane (`flex-1` / `min-h-0`) instead of a fixed 420px height.

## CI Jedi round 3 (behavior-preserving)

- Unsupported types can be **saved on edit when the type is unchanged** (import/edit parity with `buildPanelTypeGroups`).
- Create and edit type-changes **onto** an unsupported type still fail closed. Heatmap chip stays disabled until registry `ready`.

## Blast Radius

Dashboard Add panel and panel edit open this modal. Save still calls `createPanel` / `updatePanel`. Trace panels no longer send leftover PromQL as `expr`.

## Verification

`bun run test` on `PanelEditModal.spec.tsx`, `chartBuilder.spec.ts`, and `ChartBuilderPreview.spec.tsx`: 30 passed. `bun run type-check` and `bun run lint` clean.

<div><a href="https://cursor.com/agents/bc-364181a0-2897-4759-acb2-f4acf8f2ec7a?cursor_ref=pr_footer&cursor_cta=open_in_web"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-364181a0-2897-4759-acb2-f4acf8f2ec7a&cursor_ref=pr_footer&cursor_cta=open_in_cursor"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

File: frontend/src/components/PanelEditModal.tsx [hunk 1]
```diff
@@ -201,53 +204,64 @@ export function PanelEditModal({ dashboardId, panel, onClose, onSaved }: PanelEd
...
 
-    const size = isEditing && panel?.grid_pos
-      ? { w: panel.grid_pos.w, h: panel.grid_pos.h }
-      : { w: DEFAULT_GRID_POS.w, h: DEFAULT_GRID_POS.h }
+    const size =
+      isEditing && panel?.grid_pos
+        ? { w: panel.grid_pos.w, h: panel.grid_pos.h }
+        : { w: DEFAULT_GRID_POS.w, h: DEFAULT_GRID_POS.h }
+
+    const resolvedTitle = resolveChartBuilderTitle(title, queryText)
```
Comment (janhoon) [inline frontend/src/components/PanelEditModal.tsx]: **High — create-only title fallback leaks into edits.** `resolveChartBuilderTitle` runs for updates too. Clearing the visible title on edit validates against the query (or `Panel`) and silently renames the panel instead of `Title is required`.

**Fix:** only call `resolveChartBuilderTitle` when `!isEditing`. On edit, validate/save `title.trim()` and keep the required-title failure. Add a regression test that empty title on edit still errors.

File: frontend/src/components/PanelEditModal.tsx [hunk 2]
```diff
@@ -354,123 +365,133 @@ export function PanelEditModal({ dashboardId, panel, onClose, onSaved }: PanelEd
...
-                        disabled={option.disabled}
-                      >
-                        {option.label}
-                      </option>
-                    ))}
-                  </optgroup>
-                ))}
-              </select>
-            </div>
+          <div className="sr-only">
```
Comment (janhoon) [inline frontend/src/components/PanelEditModal.tsx]: **High — chart-builder chrome replaces the real type registry for sighted users.** Only four chips are visible; the full `panelTypeGroups` select is `sr-only`. Heatmap is selectable while the registry labels it unsupported. Editing gauge/pie/logs/traces/text shows no selected chip and `chartBuilderTypeLabel` lies with “Time series”.

**Prefer judo:** gate this chip strip (and Figma Main layout) to create / `isChartBuilderType`. For other types keep a visible type control. Disable or omit Heatmap until the registry says supported. Never auto-remap an existing non-chip type to Time series in the preview.

File: frontend/src/components/PanelEditModal.tsx [hunk 3]
```diff
@@ -354,123 +365,133 @@ export function PanelEditModal({ dashboardId, panel, onClose, onSaved }: PanelEd
...
+                    </option>
+                  ))}
+                </optgroup>
+              ))}
+            </select>
           </div>
 
           {needsDatasource && datasources.length > 0 ? (
-            <div className="mb-5">
+            <div className={hideDatasourcePicker ? 'sr-only' : undefined}>
```
Comment (janhoon) [inline frontend/src/components/PanelEditModal.tsx]: **High — datasource picker hidden while CTA claims Victoria/prod.** `hideDatasourcePicker` puts the only provider control in `sr-only` for chart-builder types. Create often saves with empty `datasource_id` (default Prometheus path) while the button still says `Pin to Victoria / prod`.

**Fix:** show the datasource control whenever the user can choose (at least when `datasources.length > 0` on create/edit), and derive the pin label from the selected datasource name/type (fallback honest copy like `Pin panel`) instead of a hard-coded Victoria string.

File: frontend/src/components/ChartBuilderPreview.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        className="flex h-[420px] w-full items-end gap-[3px] overflow-hidden"
+        data-testid="chart-builder-preview-bars"
+      >
+        {CHART_BUILDER_PREVIEW_BARS.map((bar) => (
+          <div
+            key={bar.id}
+            className="min-w-0 flex-1 rounded-[2px]"
+            style={{
+              height: `${bar.height}px`,
+              backgroundColor: 'var(--color-primary)',
```
Comment (janhoon) [inline frontend/src/components/ChartBuilderPreview.tsx]: **Medium — preview series uses brand primary.** Chart data should use the visualization palette (Steel Blue default per AGENTS.md); brand brass/primary is for emphasis/thresholds, not series fill.

**Fix:** paint bars with the viz default token (e.g. Steel Blue / chart series token), not `var(--color-primary)`. Update the spec assertion accordingly.

File: frontend/src/components/PanelEditModal.tsx [hunk 5]
```diff
@@ -354,123 +365,133 @@ export function PanelEditModal({ dashboardId, panel, onClose, onSaved }: PanelEd
...
+          Close
+        </button>
+        <button
+          type="button"
+          className="sr-only"
+          data-testid="panel-edit-cancel-btn"
+          onClick={onClose}
         >
+          Cancel
+        </button>
```
Comment (janhoon) [inline frontend/src/components/PanelEditModal.tsx]: **Medium — no visible dismiss.** Close/Cancel are `sr-only` on a full-screen editor. Escape still works, but sighted users have no on-screen exit without submitting.

**Fix:** restore a visible close/cancel control (match Figma if it has one; otherwise a clear Cancel / X in the config column). Keep Escape.

**Comment by janhoon** [comment]
CI Jedi thermo-nuclear fix round 1 launched on `cursor/refresh-chart-builder-12-73-ec7a` @ `f2b18213a9eddb36c308ab785cec5209f62055d1`.

**Review by janhoon — COMMENTED** [review]
## CI Jedi thermo-nuclear review

**Verdict: OPEN** — ambitious Figma create chrome landed on the shared panel editor and regressed edit/authoring paths that this PR did not claim to redesign.

### What’s good
- `chartBuilder.ts` is a clean constants/helpers module with focused specs.
- Create-path copy, default PromQL, stacked type chips, and static preview match the stated Figma 12:73 intent.
- Trace leftover PromQL no longer leaks into `expr` (`!isBuiltinTracePanel`) — real blast-radius care.
- Escape-to-close is a welcome addition.

### Actionable findings (5)
1. **High — create-only title fallback applied to updates** — clearing title on edit silently renames the panel to the query / `Panel` instead of failing validation.
2. **High — chart-builder UI forced onto every panel type** — four chips + `sr-only` full select hide gauge/pie/logs/traces/text; Heatmap is clickable while registry marks it unsupported; editing a non-chip type shows no selected chip and a lying “Time series” preview label.
3. **High — datasource authoring hidden while CTA hardcodes Victoria/prod** — chart-builder types bury the picker in `sr-only`; pin copy does not track the actual save target.
4. **Medium — no visible dismiss** — Close/Cancel are `sr-only`; Escape works, but sighted users have no on-screen way out without pinning.
5. **Medium — design-system / layout debt from the restyle** — preview bars use brand primary instead of viz palette; 360px + preview side-by-side clips on narrow viewports; pin text uses hard-coded `#0B0D0F`.

### Prefer code-judo over polish
Gate the Figma Main layout to **create** (and optionally edit of `CHART_BUILDER_TYPES`). Do not pretend the four-chip chrome is the universal panel editor. Align pin CTA with the real datasource. Fail closed on unsupported Heatmap.

Fix agent launching for round 1 (`bc-4c27a1b8-e876-42ea-becd-0d0f912839b7`). Never merging from CI Jedi.

**Review by janhoon — COMMENTED** [review]
## CI Jedi thermo-nuclear re-review

**Head:** `f6ee41684bf6b4441e0201e3f43fac63dc381a06` (prior reviewed `f2b18213`)
**Verdict: still OPEN** — this push is test-only (`test(chart-builder): lock Figma 12:73 designer notes`). It locks the 360px column, selected-chip accent stroke, ink-on-gold pin, and undecorated preview. It does **not** address the product findings.

### Still open (5)
1. **High** — create title fallback (`resolveChartBuilderTitle`) still runs on edit / updates.
2. **High** — chart-builder four-chip chrome + `sr-only` full type select still forced onto all panel types; Heatmap selectable while registry marks unsupported; non-chip edit types get lying “Time series” preview label.
3. **High** — datasource picker still `sr-only` for chart-builder types while CTA hardcodes `Pin to Victoria / prod`.
4. **Medium** — Close/Cancel still `sr-only`; no visible dismiss (Escape only).
5. **Medium** — preview bars still `var(--color-primary)` instead of viz palette; hard-coded `#0B0D0F` pin text; narrow layout still side-by-side 360px.

Round marker stays at `<!-- ci-jedi-tn-round: 1 -->`. Fix agent `bc-4c27a1b8-e876-42ea-becd-0d0f912839b7` is still running — **not** launching a second agent or bumping the round. Steered it onto this head so it lands fixes on top of the test locks.

Never merging from CI Jedi. PMO owns merge.

**Review by janhoon — COMMENTED** [review]
## CI Jedi thermo-nuclear re-review

**Head:** `e5bceb22a9b7ddb6c3b9ee8574a4d6d952a82c87` (prior reviewed `f6ee4168` / `f2b18213`)
**Verdict: still OPEN** — round 1 landed the right product gates; a few edit-path regressions remain.

### Round 1 addressed (good)
- Title fallback is create-only; empty title on edit fails closed with tests.
- Chip chrome gated; Heatmap disabled + save fails closed; non-builder edit keeps a visible type select and honest preview label.
- Datasource picker visible; pin copy follows the selected datasource.
- Close/Cancel visible; Escape retained.
- Preview series uses `--color-viz-0`; ink-on-gold via `--primary-foreground`; columns stack below ~425px.

### Still actionable (3)
1. **High — PromQL `QueryBuilder` removed for edit (and create)** — bare single-line `<input>` replaces Monaco for every non-signal PromQL path. Figma bare field is fine for **create**; restoring `QueryBuilder` on **edit** (at least) preserves authoring for real queries.
2. **Medium — edit save CTA still speaks “Pin”** — `chartBuilderPinLabel(...)` runs even when `isEditing`. Gauge/logs edit should say `Save Changes` (or equivalent); keep pin copy for create only.
3. **Medium — editing a builder type still traps type migration** — when chips show, the full registry select stays `sr-only`, so sighted users cannot change `line_chart` → gauge/logs/etc. Keep chips for the Figma set, but leave a visible escape hatch to the full type control on edit.

Nit (non-blocking): preview bars container is fixed `h-[420px]` instead of filling the remaining pane.

Fix agent round 2 launching on the same branch. Never merging from CI Jedi. PMO owns merge.

File: frontend/src/components/PanelEditModal.tsx [hunk 6]
```diff
@@ -547,14 +594,24 @@ export function PanelEditModal({ dashboardId, panel, onClose, onSaved }: PanelEd
...
                 <ElasticsearchQueryEditor
                   value={queryText}
                   onChange={setQueryText}
                   signal={querySignal === 'traces' ? 'metrics' : querySignal}
                   disabled={loading}
                   onSignalChange={handleNonTraceSignalChange}
                 />
+              ) : (
+                <input
+                  id="promql-query"
```
Comment (janhoon) [inline frontend/src/components/PanelEditModal.tsx:616]: **High — PromQL QueryBuilder removed for edit.** This bare single-line input replaces Monaco for every default PromQL path (create and edit). Figma 12:73 bare field is fine for **create**; on **edit**, restore `QueryBuilder` (or at least a multi-line capable editor) so real PromQL authoring is not a capability regression. Keep the create chrome thin.

File: frontend/src/components/PanelEditModal.tsx [hunk 7]
```diff
@@ -614,39 +671,33 @@ export function PanelEditModal({ dashboardId, panel, onClose, onSaved }: PanelEd
...
               style={{
-                background:
-                  'linear-gradient(135deg, var(--color-primary), var(--color-primary-dim))',
+                backgroundColor: 'var(--color-primary)',
+                color: 'var(--primary-foreground)',
               }}
               disabled={loading}
             >
-              {loading ? 'Saving...' : isEditing ? 'Save Changes' : 'Add Panel'}
+              {loading ? 'Saving...' : chartBuilderPinLabel(selectedDatasource)}
```
Comment (janhoon) [inline frontend/src/components/PanelEditModal.tsx]: **Medium — edit CTA still says Pin.** `chartBuilderPinLabel` runs for updates too, so editing a gauge/logs panel still reads like pinning. Use pin copy only when `!isEditing`; on edit use `Save Changes` (or the previous edit label).

File: frontend/src/components/PanelEditModal.tsx [hunk 8]
```diff
@@ -354,123 +379,147 @@ export function PanelEditModal({ dashboardId, panel, onClose, onSaved }: PanelEd
...
+                    }}
+                  >
+                    {entry.label}
+                  </button>
+                )
+              })}
             </div>
+          ) : null}
+
+          <div className={showChartBuilderTypeChips ? 'sr-only' : 'flex flex-col gap-2'}>
```
Comment (janhoon) [inline frontend/src/components/PanelEditModal.tsx]: **Medium — builder-type edit traps type migration.** While chips are visible the full `panelTypeGroups` select stays `sr-only`, so a sighted user editing `line_chart` cannot switch to gauge/logs/text. Keep the Figma chip strip, but on edit leave a visible escape hatch to the registry select (below chips or behind a “More types” control).

**Review by janhoon — COMMENTED** [review]
Inline follow-ups for the three remaining findings on `e5bceb22`.

**Comment by janhoon** [comment]
CI Jedi thermo-nuclear fix round 2 launching on `cursor/refresh-chart-builder-12-73-ec7a` @ `e5bceb22a9b7ddb6c3b9ee8574a4d6d952a82c87`.

**Review by janhoon — COMMENTED** [review]
## CI Jedi thermo-nuclear re-review

**Head:** `c35eb7b92c442d9b77027ca0132b1f4617f537cc` (prior reviewed `e5bceb22` / `b7e2515d` was knip-only; HEAD moved mid-run to this round-2 fix)
**Verdict: CLEAN** — round 2 closes the three remaining edit-path findings without undoing Figma 12:73 create chrome.

### Prior remaining @ e5bceb22 — closed
1. **QueryBuilder on edit** — default PromQL path uses `<QueryBuilder>` when `isEditing`; create keeps the bare `promql-query` field. Specs lock create has no `mock-query-builder` and edit does.
2. **Edit CTA must not say Pin** — submit label is `Save Changes` on edit; `chartBuilderPinLabel(...)` is create-only and still tracks the selected datasource.
3. **Visible type escape hatch** — `hideFullTypeSelect = showChartBuilderTypeChips && !isEditing`; editing `line_chart`/`stat`/`table` keeps chips **and** a visible registry select (migration to gauge covered in spec). Create still `sr-only`s the full select behind chips.

### Round 1 items — still closed (no regression)
Title fallback create-only; Heatmap chip disabled + save fails closed; non-builder edit keeps honest type chrome/preview; datasource picker visible; Close/Cancel visible; `--color-viz-0` preview; ink-on-gold via `--primary-foreground`; columns stack below ~425px.

### Also good
Prior nit addressed: preview bars use `flex-1` / `min-h-0` instead of fixed `h-[420px]`. Explore still on Monaco — dropping the unused Explore `QueryBuilder` mock is correct.

### Non-blocking note (not reopening)
`panelTypeUnsupported` still fail-closes save for registry-unsupported types on **edit** as well as create (Bugbot). Acceptable under the round-1 Heatmap fail-closed judo for this scoped restyle; PMO can carve import-edit parity later if needed.

Quality bar met for CI Jedi. **No further fix agent.** Never merging from CI Jedi. PMO owns merge.

**Review by janhoon — COMMENTED** [review]
## CI Jedi thermo-nuclear re-review

**Head:** `c35eb7b92c442d9b77027ca0132b1f4617f537cc` (prior reviewed `e5bceb22`)
**Verdict: still OPEN** — round 2 landed the edit-path restorations; one regression remains from the fail-closed gate.

### Round 2 addressed (good)
- Edit PromQL path restores `QueryBuilder`; create keeps the Figma bare input.
- Edit save CTA is `Save Changes`; pin copy is create-only via `chartBuilderPinLabel`.
- Editing a chart-builder type keeps chips **and** a visible full type select (`hideFullTypeSelect = chips && !isEditing`).
- Preview pane uses `flex-1` / `min-h-0` instead of fixed `420px`.

### Still actionable (1)
1. **Medium — unsupported-type edit saves fail closed too broadly** — `isUnsupportedPanelType` + the `handleSubmit` gate reject every `supportStatus === 'unsupported'` type on **edit**, not just Heatmap on create. That conflicts with `buildPanelTypeGroups`, which keeps those types selectable so existing dashboards can open them for edit/import parity (panel renders empty state). Changing title/query on an existing heatmap/scatter (etc.) now errors instead of saving.

**Fix:** fail closed only when **selecting** an unsupported type that should not be created (create path, or a type **change** onto unsupported). If `isEditing && panelType === panel.type`, allow the save. Keep Heatmap chip disabled + create fail-closed. Add a regression test for edit of an existing unsupported type.

Nit (non-blocking): preview bars still use fixed px heights inside the flex pane — fine for Figma chrome.

Fix agent round 3 launching on the same branch. Never merging from CI Jedi. PMO owns merge.

File: frontend/src/components/PanelEditModal.tsx [hunk 9]
```diff
@@ -261,6 +285,12 @@ export function PanelEditModal({ dashboardId, panel, onClose, onSaved }: PanelEd
       return
     }
 
+    if (panelTypeUnsupported) {
+      const unsupportedLabel = chartBuilderTypeLabel(panelType)
+      setError(`${unsupportedLabel} is not supported`)
+      return
+    }
```
Comment (janhoon) [inline frontend/src/components/PanelEditModal.tsx]: **Medium — unsupported-type edit saves fail closed too broadly.**

This gate rejects every `supportStatus === 'unsupported'` type on edit, not just Heatmap on create. That fights `buildPanelTypeGroups`, which keeps unsupported types selectable for edit/import parity.

**Fix:** only fail closed when creating, or when the user **changes** type onto an unsupported value. If `isEditing && panelType === panel.type`, allow the save. Keep the Heatmap chip disabled. Add a regression test for editing an existing unsupported panel (title/query change still saves).

**Review by janhoon — COMMENTED** [review]
Inline follow-up for the remaining unsupported-edit save gate on `c35eb7b9`.

**Comment by janhoon** [comment]
CI Jedi thermo-nuclear fix round 3 launching on `cursor/refresh-chart-builder-12-73-ec7a` @ `c35eb7b92c442d9b77027ca0132b1f4617f537cc`.

**Review by janhoon — COMMENTED** [review]
## CI Jedi thermo-nuclear re-review

**Head:** `dacebd762e7873ae991bf04d1928af40ba4ce337` (prior gated `c35eb7b9` OPEN)
**Verdict: CLEAN** — round 3 closes the last residual without undoing Figma 12:73 create chrome or the round 1–2 edit-path restorations.

### Prior residual @ c35eb7b — closed
1. **Unsupported-type edit saves** — `handleSubmit` now fails closed only when `panelTypeUnsupported && !(isEditing && panelType === panel?.type)`. Edit of an existing heatmap (type unchanged) saves title/query; create and edit type-changes onto unsupported still error. Specs lock both paths. Heatmap chip stays `disabled` until registry `ready`.

### Rounds 1–2 — still closed (no regression)
- Title fallback create-only; empty title on edit fails closed.
- Chip chrome gated; non-builder edit keeps visible type select + honest preview label.
- Datasource picker visible; pin copy create-only via `chartBuilderPinLabel`.
- Edit PromQL restores `QueryBuilder`; create keeps bare `promql-query` field.
- Edit CTA is `Save Changes`; editing builder types keeps chips **and** visible full select (`hideFullTypeSelect = chips && !isEditing`).
- Close/Cancel visible; Escape retained; `--color-viz-0` preview; ink-on-gold via `--primary-foreground`; columns stack below ~425px; preview pane `flex-1` / `min-h-0`.

### Also good
r3 delta is surgical (1 gate line + 2 regression tests). Aligns with `buildPanelTypeGroups` import/edit parity comment. Trace leftover PromQL still gated by `!isBuiltinTracePanel`.

### Non-blocking notes (not reopening)
- Create title field remains `sr-only` (Figma intent; pin uses query-as-title).
- Preview bars still use fixed px heights inside the flex pane — fine for chrome.

Quality bar met for CI Jedi. **MAX FIX ROUNDS N=3 already; no further fix agent.** Never merging from CI Jedi. PMO owns merge.

**Review by janhoon — COMMENTED** [review]
## CI Jedi thermo-nuclear re-review

**Head:** `dacebd762e7873ae991bf04d1928af40ba4ce337` (prior reviewed `c35eb7b9`)
**Verdict: CLEAN** — round 3 closes the unsupported-edit save gate without undoing Figma 12:73 create chrome or the round-1/2 edit restorations.

### Prior remaining @ c35eb7b9 — closed
1. **Unsupported-type edit saves** — `handleSubmit` now fails closed only when creating or **changing onto** an unsupported type: `panelTypeUnsupported && !(isEditing && panelType === panel?.type)`. Existing heatmap (etc.) can still save title/query. Specs lock create fail-closed, unchanged-edit save, and edit type-change onto Heatmap fail-closed.

### Prior rounds — still closed (no regression)
- Title fallback create-only; empty edit title fails validation.
- Chip chrome gated; Heatmap chip disabled until registry `ready`; non-builder edit keeps honest type chrome/preview.
- Datasource picker visible; pin copy create-only via `chartBuilderPinLabel`.
- Close/Cancel visible; Escape retained.
- Edit PromQL restores `QueryBuilder`; create keeps bare input.
- Edit CTA is `Save Changes`; chips + visible type select on builder-type edit.
- Preview uses `--color-viz-0` / flex pane fill; ink-on-gold via `--primary-foreground`; columns stack below ~425px.

### Non-blocking notes (not reopening)
- `isUnsupportedPanelType` still special-cases heatmap as `status !== 'ready'` vs others `=== 'unsupported'` — intentional for the Figma chip until registry marks ready.
- `PanelEditModal` remains a large shared editor; this PR correctly scopes chrome rather than splitting modules — fine for this restyle.

Quality bar met for CI Jedi. Round marker already `<!-- ci-jedi-tn-round: 3 -->`. **No further fix agent.** Never merging from CI Jedi. PMO owns merge.
